### ISSUE TITLE
gvcf: DP sometimes below MIN_DP

### ISSUE THREAD

**Issue by mlin** [freebayes/freebayes#239]
I generated a gvcf (with v1.0.2, --gvcf-chunk 100) on a whole-chromosome BAM and plotted the following distribution of DP and MIN[_DP] in all the reference confidence records:

![](https://pbs.twimg.com/media/CZ310cGXEAEFUrn.jpg:large)

Curiously there's a small number of records (less than 0.1%) below the diagonal, or DP < MIN_DP which seems wrong.

As a trivial example, this is the first emitted record:

```
12      60001   .       C       <*>     0       .       DP=0;END=60102;MIN=1    GQ:DP:MIN:QR:QA 2502:0:1:2502:-0
```

**Comment by mlin** [comment]
@mlin to provide a reproduction

**Comment by mlin** [comment]
@ekg @zeeev okay I have a compact reproduction in the current tip of master (41c1313)

* BAM: https://dl.dnanex.us/F/D/F15ygPgq4pb8f69kBv0p8gB9Fx656V67fx1PFJKK/NA12878_chr21_10137059_10137232.bam
* reference genome: https://dl.dnanex.us/F/D/vxF7Bk5b90vbF3VyK403B6FpbG424vBgg39fYV6k/ucsc_hg19.fa.gz

invocation (nb I am using --gvcf-chunk):

```
freebayes -f ./genome/ucsc_hg19.fa -b /home/dnanexus/in/sorted_bams/0/NA12878_chr21_10137059_10137232.bam -0 --gvcf --gvcf-chunk 100
```

* VCF: https://dl.dnanex.us/F/D/FggqQzX3Qq39x4Xq13GyB6GGxxkpKgVK5vkyxXJz/NA12878_chr21_10137059_10137232.vcf.gz

The VCF contains two records with DP < MIN_DP, excerpted:

```
...
chr21   10137094        .       T       <*>     0       .       DP=153;END=10137129;MIN_DP=177  GQ:DP:MIN_DP:QR:QA      202431:153:177:202535:104.158
...
chr21   10137461        .       G       <*>     0       .       DP=0;END=10137485;MIN_DP=1      GQ:DP:MIN_DP:QR:QA      922:0:1:922:-0
```

If I run all of chromosome 21, there are 736 records with DP < MIN_DP. A large majority are DP=0,MIN_DP=1 but there are many examples of the (even more disturbing) well-populated case.

* chromosome 21 BAM: https://dl.dnanex.us/F/D/2V6y9yJxzby14KyYVP1147V6F5q0zG1847JXGF9k/NA12878_chr21.bam
* chromosome 21 VCF: https://dl.dnanex.us/F/D/fGfqj0yF5GZZg12JPjk2K3BvXzXZXYYgv1Q17Z1X/NA12878_chr21.vcf.gz

Crappy incantation to count buggy records:

```
curl https://dl.dnanex.us/F/D/fGfqj0yF5GZZg12JPjk2K3BvXzXZXYYgv1Q17Z1X/NA12878_chr21.vcf.gz \
    | zcat | cut -f8 | grep END | grep -v \# | cut -d '=' -f 2,4 --output-delimiter " " | tr -d ";END" \
    | awk '$1 < $2 {print;}' | wc -l
```

Good hunting!

**Comment by zeeev** [comment]
@mlin @ekg this bug only occurs when the standard filters are applied.  @ekg Can you comment on the nature of lines 16-22 in NonCall.cpp?

**Comment by pjotrp** [comment]
I think this was fixed with #299. Reopen if not.
